lib/cosmos/streams/tcpip_socket_stream.rb

```diff
@@ -64,13 +66,13 @@ def read
           end
         rescue IOError, Errno::ENOTSOCK
           # These can happen with the socket being closed while waiting on select
-          data = ''
+          @data = BLANK
```

Comment:
We're just trying to return a blank string which can be a constant blank string instead of a newly allocated blank string each time. When this is called again, @data it is filled with a new chunk from the socket.